packages/storage/src/indexed-db/indexed-db.test.ts

```diff
@@ -255,7 +253,7 @@ describe('IndexedDb', () => {
       await db.saveTransactionInfo(transactionInfo);
 
       const savedTransaction = await db.getTransaction(
-        new NoteSource({ inner: transactionInfo.id!.hash }),
+        new CommitmentSource_Transaction({ id: transactionInfo.id!.inner }),
```

Comment:
okay, they are equivalent? is there a reason we're using CommitmentSource_Transaction to index transactions rather than the semantically clearer TransactionId?

Reply:
indexing by (nullable) transaction ID seems the cleanest, the point of the CommitmentSource is to be able to tell whether funds come from a transaction or not.

eventually we will want to figure out the best way to index other kinds of sources (particularly IBC transfers) but for the moment the tx hash seems like the cleanest thing